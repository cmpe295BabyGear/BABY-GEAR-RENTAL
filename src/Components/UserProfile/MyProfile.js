import React, { useEffect } from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import CssBaseline from '@mui/material/CssBaseline';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { makeStyles } from '@material-ui/core/styles';
import { CardActionArea } from '@mui/material';
import {Link as RouterLink} from 'react-router-dom'
import FormatListBulletedIcon from '@mui/icons-material/FormatListBulleted';
import PaymentIcon from '@mui/icons-material/Payment';
import HomeIcon from '@mui/icons-material/Home';
import PasswordIcon from '@mui/icons-material/Password';
import HistoryIcon from '@mui/icons-material/History';
import BookmarksIcon from '@mui/icons-material/Bookmarks';
import FavoriteIcon from '@mui/icons-material/Favorite';

const theme = createTheme();
const useStyles = makeStyles({
  card: {
    maxWidth: 300,
  },
  cardmedia: {
    height: 140,
  },
})
// To-do: CHANGE THE ICON PICTURES 
export default function MyProfile () {
  const [userName, setUserName] = React.useState('test user')
  const classes = useStyles()
  useEffect(() => {
    // To- do : get and set username
    var sessionDetails = JSON.parse(sessionStorage.getItem('custId'));
  }, [])
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <main>
        <Box sx={{ flexGrow: 1 }}>
          <Container maxWidth='sm'>
            <Typography
              component='h1'
              variant='h2'
              align='center'
              color='text.primary'
              gutterBottom
            >
              Hi {userName}
            </Typography>
          </Container>
        </Box>
        <Container sx={{ py: 8 }} maxWidth='md'>
          <Grid container spacing={2}>
            <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/userAddress'>
                  {/* <CardMedia
                    // component='img'
                    component='div'
                    className={classes.cardmedia}
                    // image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/listings.png'
                    src={<FormatListBulletedIcon />}
                    alt='My listings'
                  >
                    {/* <FormatListBulletedIcon color='secondary' fontsize='large' align='center' height='140'/> */}
                  {/* </CardMedia> */}
                  <CardContent sx={{ flexGrow: 1 }}>
                    <Typography gutterBottom variant='h6' component='h2'>
                      <FormatListBulletedIcon color='secondary' fontSize='large' gutterBottom /> My Listings
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
            {/* <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/paymentOptions'>
                  {/* <CardMedia
                    component='img'
                    className={classes.cardmedia}
                    image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/PaymentOptions.png'
                    alt='payment options'
                  /> */}
                  {/* <CardContent >
                    <Typography gutterBottom variant='h6' component='h2'>
                      <PaymentIcon color='secondary' fontSize='large' gutterBottom /> Payment Options
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>  */}
            <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/userAddress'>
                  {/* <CardMedia
                    component='img'
                    className={classes.cardmedia}
                    image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/Address.png'
                    alt='addresses'
                  /> */}
                  <CardContent sx={{ flexGrow: 1 }}>
                    <Typography gutterBottom variant='h6' component='h2'>
                      <HomeIcon color='secondary' fontSize='large' /> My Addresses
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
            <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/userAddress'>
                  {/* <CardMedia
                    component='img'
                    className={classes.cardmedia}
                    image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/myorders.png'
                    alt='my orders'
                  /> */}
                  <CardContent sx={{ flexGrow: 1 }}>
                    <Typography gutterBottom variant='h6' component='h2'>
                      <HistoryIcon fontSize='large' color='secondary' /> My Orders
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
            <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/userAddress'>
                  {/* <CardMedia
                    component='img'
                    className={classes.cardmedia}
                    image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/wishlist.png'
                    alt='wishlist'
                  /> */}
                  <CardContent sx={{ flexGrow: 1 }}>
                    <Typography gutterBottom variant='h6' component='h2'>
                      <FavoriteIcon fontSize='large' color='secondary'/> My WishList
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
            <Grid item xs={12} sm={6} md={4}>
              <Card className={classes.card}>
                <CardActionArea underline='none' component={RouterLink} to='/changePwd'>
                  {/* <CardMedia
                    component='img'
                    className={classes.cardmedia}
                    image='https://prelovedbabyitems.s3.us-east-2.amazonaws.com/OtherPictures/password.png'
                    alt='security'
                  /> */}
                  <CardContent sx={{ flexGrow: 1 }}>
                    <Typography gutterBottom variant='h6' component='h2'>
                      <PasswordIcon fontSize='large' color='secondary' /> Change Password
                    </Typography>
                  </CardContent>
                </CardActionArea>
              </Card>
            </Grid>
          </Grid>
        </Container>
      </main>
    </ThemeProvider>
  )
}
