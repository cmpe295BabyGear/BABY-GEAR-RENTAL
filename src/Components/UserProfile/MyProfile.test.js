import React from "react";
import { render, cleanup } from "@testing-library/react";
import MyProfile from "./MyProfile";

import {
  BrowserRouter as Router,
} from "react-router-dom";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Router><MyProfile /></Router>);
  expect(asFragment()).toMatchSnapshot();
});