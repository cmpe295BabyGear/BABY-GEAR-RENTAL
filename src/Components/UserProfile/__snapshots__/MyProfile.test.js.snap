// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-bp5fon-MuiStack-root"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              color="inherit"
              href="/"
              underline="hover"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              color="inherit"
              href="/myProfile"
              underline="hover"
            >
              My Profile
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <main>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
          >
            My Profile
          </h1>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd css-jcfe1v-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                href="/myListings"
                tabindex="0"
                underline="none"
              >
                <div
                  class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeLarge css-1nryff-MuiSvgIcon-root"
                      data-testid="FormatListBulletedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 10.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-6c-.83 0-1.5.67-1.5 1.5S3.17 7.5 4 7.5 5.5 6.83 5.5 6 4.83 4.5 4 4.5zm0 12c-.83 0-1.5.68-1.5 1.5s.68 1.5 1.5 1.5 1.5-.68 1.5-1.5-.67-1.5-1.5-1.5zM7 19h14v-2H7v2zm0-6h14v-2H7v2zm0-8v2h14V5H7z"
                      />
                    </svg>
                     My Listings
                  </h2>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                href="/userAddress"
                tabindex="0"
                underline="none"
              >
                <div
                  class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeLarge css-1nryff-MuiSvgIcon-root"
                      data-testid="HomeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                      />
                    </svg>
                     My Addresses
                  </h2>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                href="/myOrders"
                tabindex="0"
                underline="none"
              >
                <div
                  class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeLarge css-1nryff-MuiSvgIcon-root"
                      data-testid="HistoryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
                      />
                    </svg>
                     My Orders
                  </h2>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                href="/changePwd"
                tabindex="0"
                underline="none"
              >
                <div
                  class="MuiCardContent-root css-2r3irp-MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeLarge css-1nryff-MuiSvgIcon-root"
                      data-testid="PasswordIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2 17h20v2H2v-2zm1.15-4.05L4 11.47l.85 1.48 1.3-.75-.85-1.48H7v-1.5H5.3l.85-1.47L4.85 7 4 8.47 3.15 7l-1.3.75.85 1.47H1v1.5h1.7l-.85 1.48 1.3.75zm6.7-.75 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H15v-1.5h-1.7l.85-1.47-1.3-.75L12 8.47 11.15 7l-1.3.75.85 1.47H9v1.5h1.7l-.85 1.48zM23 9.22h-1.7l.85-1.47-1.3-.75L20 8.47 19.15 7l-1.3.75.85 1.47H17v1.5h1.7l-.85 1.48 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H23v-1.5z"
                      />
                    </svg>
                     Change Password
                  </h2>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
